substrate/frame/staking-async/src/pallet/mod.rs

```diff
@@ -2129,14 +2131,16 @@ pub mod pallet {
 		) -> DispatchResult {
 			ensure_signed(origin)?;
 			let min_commission = MinCommission::<T>::get();
-			Validators::<T>::try_mutate_exists(validator_stash, |maybe_prefs| {
-				maybe_prefs
-					.as_mut()
-					.map(|prefs| {
-						(prefs.commission < min_commission)
-							.then(|| prefs.commission = min_commission)
-					})
-					.ok_or(Error::<T>::NotStash)
+			Validators::<T>::try_mutate_exists(&validator_stash, |maybe_prefs| {
```

Comment:
can you rewrite this with a normal `let mut validator = Validators::<T>::get().ok_or(Error::<T>::NotStash.into()); mutate it ; Validators::<T>::put()`? 

I think it will be more readable :)